### ISSUE TITLE
Reflections and Refractions

### ISSUE THREAD

**Issue by Midda-C** [huwb/crest-oceanrender#3]
Hi, you mention that the dynamics branch features reflections and refraction, but I don't see either.  Is there something specific which needs to be done to enable these?

Thanks.

**Comment by huwb** [comment]
Hi Midda! These refer to the wave dynamics. For example, waves will reflect off obstacles. The effect is quite subtle but I could show it in a screenshot of the wave dynamics buffer if it helps.

Let me know if this is what you are after - I suspect you are referring to reflection and refraction of light, i.e. a transparent ocean surface? I haven't got around to making it support transparency but I will bump that to the top of my list.

**Comment by Midda-C** [comment]
Ah, that makes more sense now.  Yes, I was thinking of refraction of light.

We (I work with Dizzy who posted about tessellation) are investigating ocean solutions, and Crest has fantastic motion, which is an important factor for our project.  We do though need an ocean sim which has transparency and would work with a day/night cycle, but if this is something you were planning to add anyway, that would be great.

Fantastic work, Crest looks amazing!

**Comment by huwb** [comment]
I finally gave the shading some much needed love. It now supports transparency and a simple refraction (I believe I'm doing it the "right" unity way, it didnt require any drastic changes). I also improved the subsurface scattering. I put a big old ugly cube in the world to demo it for now, but I'll consider replacing it with something else.

I'm optimistic about supporting a day/night cycle (by manually setting up different ocean materials for each time of day), but if there are other requirements I'd be happy to hear them.

I've added 2 other improvement ideas to the readme once it has some island geometry. One is to dampen waves in shallow water in a physically accurate way (dampen big wavelengths first, then small wavelengths), which i think will give more interesting wave motion in shallow water than just scaling down the overall waves. Second is supporting a world aligned colour map to give nice shallow water colours. These will be fairly easy to implement and should make a big impact.

For now I will close this issue, thanks again for the message and let me know if you do decide to use Crest!
